libs/Microsoft.MixedReality.WebRTC/Interop/PeerConnectionInterop.cs

```diff
@@ -814,31 +824,34 @@ internal struct RemoteVideoTrackAddedInfo
         #endregion
 
         #region Utilities
+
         class RemoteDescArgs
         {
-            public ActionDelegate callback;
-            public ManualResetEventSlim completedEvent;
+            public PeerConnectionRemoteDescriptionAppliedDelegate callback;
+            public TaskCompletionSource<bool> tcs;
         }
 
         public static Task SetRemoteDescriptionAsync(PeerConnectionHandle peerHandle, string type, string sdp)
         {
-            return Task.Run(() =>
+            var args = new RemoteDescArgs
+            {
+                callback = RemoteDescriptionApplied,
+                tcs = new TaskCompletionSource<bool>() // dummy bool, unused
+            };
+
+            Task.Run(() =>
```

Comment:
You don't need this Task anymore, the request can be started synchronously.

Reply:
You're probably right. Trying to think of a reason to keep it on a background .NET worker thread, but I don't think there's any for that one.